calc/vxlan_resolver.go

```diff
@@ -432,19 +441,37 @@ func (c *VXLANResolver) routeWithinVXLANPool(r vxlanRoute) bool {
 }
 
 // routesFromBlock returns a list of routes which should exist based on the provided
-// allocation block. Right now, we only support strict affinity so this will always
-// be the block's CIDR.
+// allocation block.
 func (c *VXLANResolver) routesFromBlock(blockKey string, b *model.AllocationBlock) map[string]vxlanRoute {
+	routes := make(map[string]vxlanRoute)
+
+	for _, alloc := range b.NonAffineAllocations() {
+		if alloc.Host == "" {
+			logrus.WithField("IP", alloc.Addr).Warn(
+				"Unable to create VXLAN route for IP; the node it belongs to was not recorded")
+			continue
+		}
+		r := vxlanRoute{
+			dst:  ip.CIDRFromNetIP(alloc.Addr.IP),
+			node: alloc.Host,
+		}
+		routes[r.Key()] = r
+	}
+
 	if b.Host() == c.hostname {
 		logrus.Debug("Skipping VXLAN routes for local node")
 		return nil
```

Comment:
I think we need to return `routes` here instead of nil, since it might contain borrowed routes from this block.

Alternatively, we just move this check down below:

```
if b.Host() != c.hostname && b.Host != "" {
  // Add the route for the block CIDR if it is for a remote node.
}
```

Reply:
Good catch

Reply:
Fixed with a new calc graph test to prove it :-)